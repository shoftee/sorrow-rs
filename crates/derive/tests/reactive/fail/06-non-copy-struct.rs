use sorrow_derive::Reactive;

#[derive(Reactive)]
struct NonCopy {
    non_copy_field: Vec<i32>,
}

fn main() {}
