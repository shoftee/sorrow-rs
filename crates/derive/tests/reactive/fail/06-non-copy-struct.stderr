error[E0507]: cannot move out of `s.non_copy_field` which is behind a shared reference
 --> tests/reactive/fail/06-non-copy-struct.rs:5:5
  |
5 |     non_copy_field: Vec<i32>,
  |     ^^^^^^^^^^^^^^^^^^^^^^^^ move occurs because `s.non_copy_field` has type `Vec<i32>`, which does not implement the `Copy` trait
